### PR TITLE
[SPARK-44338][SQL] Fix view schema mismatch error message

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

This fixes a minor issue that the view schema mismatch error message may have an extra space between view name and `AS SELECT`. It also takes the chance to simplify the view test a little bit, to always use `tableIdentifier` to produce the view name.

### Why are the changes needed?

small fix of the error message

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

existing tests

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewTestSuite.scala [hunk 1]
```diff
@@ -640,13 +630,13 @@ class PersistedViewTestSuite extends SQLViewTestSuite with SharedSparkSession {
...
           sql(s"SELECT * FROM test_view")
         }
+        val unquotedViewName = tableIdentifier("test_view").unquotedString
         checkError(
           exception = e,
           errorClass = "INCOMPATIBLE_VIEW_SCHEMA_CHANGE",
           parameters = Map(
-            "viewName" -> "`spark_catalog`.`default`.`test_view`",
-            "suggestion" ->
-              "CREATE OR REPLACE VIEW spark_catalog.default.test_view  AS SELECT * FROM t",
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewTestSuite.scala:649]: this test shows the double space issue.

**Comment by cloud-fan** [comment]
cc @gengliangwang

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala [hunk 2]
```diff
@@ -909,11 +909,11 @@ class SessionCatalog(
...
       val userSpecifiedColumns =
         if (metadata.schema.fieldNames.toSeq == metadata.viewQueryColumnNames) {
-          ""
+          " "
         } else {
-          s"(${metadata.schema.fieldNames.mkString(", ")})"
+          s" (${metadata.schema.fieldNames.mkString(", ")}) "
         }
-      Some(s"CREATE OR REPLACE VIEW $viewName $userSpecifiedColumns AS $viewText")
+      Some(s"CREATE OR REPLACE VIEW $viewName${userSpecifiedColumns}AS $viewText")
```
Comment (beliefer) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala:916]: Thank you for the fix.

**Review by beliefer — APPROVED** [review]
LGTM.

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
Thanks, merging to master

**Pull request closed without merging**